### PR TITLE
CRUD endpoint for users

### PR DESCRIPTION
**Pull request by arkes987**
Pull request refer to #9 .

Added:

- CRUD endpoint for user entity with methods described in issue.
- Tests for UserLogic and UserMapper, all mentioned in issue.

### CODE REVIEW

File: Offre.Test/User/UserLogicTests.cs [hunk 1]
```diff
@@ -12,77 +15,194 @@ namespace Offre.Test.User
...
         {
             return new UserLogic(_offreContextMock.Object);
         }
 
-        //[TestMethod]
-        //public void GetAllUsers()
-        //{
-        //    var userLogic = GetTestSubject();
+        [TestMethod]
+        public void GetAllUsers_GetAllUsersFromTable_ReturnsValidUserCount()
```
Comment (wojciechowskiapp) [inline Offre.Test/User/UserLogicTests.cs]: Test methods have to be async - in this way we skip x.Result method which is in most cases a bad idea and should be omitted.

File: Offre/Startup.cs [hunk 2]
```diff
@@ -93,6 +94,7 @@ private void ConfigureOwnLogic(IServiceCollection services)
         {
             services.AddScoped<IAuthorizeLogic, AuthorizeLogic>();
             services.AddScoped<IUserLogic, UserLogic>();
+            services.AddScoped<IUserMapping, UserMapping>();
```
Comment (wojciechowskiapp) [inline Offre/Startup.cs]: mapper service can be a singleton - performance thing.

File: Offre.Test/User/UserLogicTests.cs [hunk 3]
```diff
@@ -12,77 +15,194 @@ namespace Offre.Test.User
...
+        {
+            var userLogic = GetTestSubject();
 
+            var userList = new List<UserModel>
+            {
+                new UserModel
+                {
+                    Id = 12
```
Comment (wojciechowskiapp) [inline Offre.Test/User/UserLogicTests.cs]: If there is no reason for using a random, unknown number for id, begin from the 0 and increment it by one for each.

File: Offre/Startup.cs (hunk 2 above)
Comment (arkes987) [reply Offre/Startup.cs]: Done

File: Offre.Test/User/UserLogicTests.cs (hunk 3 above)
Comment (arkes987) [reply Offre.Test/User/UserLogicTests.cs]: Done

File: Offre.Test/User/UserLogicTests.cs (hunk 1 above)
Comment (arkes987) [reply Offre.Test/User/UserLogicTests.cs]: Done

**Pull request merged**